entrypoint/remote/archive.go

```diff
@@ -0,0 +1,153 @@
+package remote
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	log "github.com/sirupsen/logrus"
+	"io"
+	"os"
+)
+
+type CosaArchive struct {
+	CreateDirs  []string
+	IncludeDirs []string
```

Comment:
Let's add an option for bare files too, such as the `jobspec`.

Reply:
Thank you for review. My bad. `IncludeDirs` is not an appropriate variable name here. Actually, it includes all the directories and files(jobspec and buildsteps) used to create the tar ball. Do you think changing the variable name to `sources` will be better? Or separating files and directories makes more sense?

Reply:
I'd lean towards `includes`, but `sources` is fine, too.

Reply:
Make sense. Updated to use `includes`.